include/envoy/config/grpc_mux.h

```diff
@@ -16,7 +16,7 @@ namespace Config {
 #define ALL_CONTROL_PLANE_STATS(COUNTER, GAUGE)                                                    \
   COUNTER(rate_limit_enforced)                                                                     \
   GAUGE(connected_state, NeverImport)                                                              \
-  GAUGE(pending_requests, Accumulate)
+  GAUGE(pending_requests, NeverImport)
```

Comment:
note to reviewers: it is worth discussing the gauges changed to NeverImport from Accumulate in this PR. They were all needed due to the assertion added to Gauge::set(). If there are objections to these transformations we should discuss what the semantics of set() should be across hot-restart; we might need a new mode.

Reply:
I was thinking about this more; if we want to allow set() on accumulated gauges we probably need the Gauge to add a new data field for the value from the parent, and make Gauge::value() return the sum of the child & parent values. We could also make this memory-neutral when hot-restart is disabled with an alternate Gauge implementation in allocator_impl.cc.

Reply:
Why does this gauge use set though? That seems broken. I would expect this to accumulate? (It's not a crisis if it doesn't and we can TODO this to fix later to land this PR if needed.)

Reply:
i think there's a few places where the data is tracked by some other non-stat variable, and then copied into the stat from time to time.